google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -263,13 +263,20 @@ public Job waitFor(RetryOption... waitOptions) throws InterruptedException {
    *
    * @throws BigQueryException upon failure
    */
-  public QueryResponse getQueryResults(QueryResultsOption... options) {
+  public QueryResult getQueryResults(QueryResultsOption... options) {
     if (getConfiguration().getType() != Type.QUERY) {
       throw new UnsupportedOperationException(
           "Getting query results is supported only for " + Type.QUERY + " jobs");
     }
 
-    return bigquery.getQueryResults(getJobId(), options);
+    QueryResponse response = bigquery.getQueryResults(getJobId(), options);
+    if (!response.jobCompleted()) {
+      throw new IllegalStateException("the job hasn't completed yet");
+    }
+    if (response.hasErrors()) {
```

Comment:
Ditto that we could check on the errors at [job status.errors](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#status.errors).

There is a chance that any errors populated here are just warnings according to the docs

> Errors here do not necessarily mean that the job has completed or was unsuccessful.

This means I think maybe we shouldn't be throwing an exception in all cases. Let's check with the BQ eng team.

Reply:
I checked the Go implementation. You're right. Removed.